Sources/eppo/EppoClient.swift

```diff
@@ -18,34 +19,78 @@ public enum Errors: Error {
     case flagConfigNotFound
 }
 
-public class EppoClient {
-    public private(set) var apiKey: String = "";
-    public private(set) var host: String = "";
-    private var configurationStore: ConfigurationStore;
-    private var assignmentCache: AssignmentCache?;
+actor EppoClientState {
+    private(set) var isLoaded: Bool = false
     
+    func checkAndSetLoaded() -> Bool {
+        if !isLoaded {
+            isLoaded = true
+            return false
+        }
+        return true
+    }
+}
+
+public class EppoClient {
     public typealias AssignmentLogger = (Assignment) -> Void
-    public var assignmentLogger: AssignmentLogger?
     
-    public init(
-        _ apiKey: String,
+    private static var instance: EppoClient?
+    
+    private(set) var apiKey: String
+    private(set) var host: String
+    private(set) var assignmentLogger: AssignmentLogger?
+    private(set) var assignmentCache: AssignmentCache?
+    private(set) var configurationStore: ConfigurationStore
+    
+    private let state = EppoClientState()
+    
+    private init(
+        apiKey: String,
         host: String = "https://fscdn.eppo.cloud",
         assignmentLogger: AssignmentLogger? = nil,
         assignmentCache: AssignmentCache? = InMemoryAssignmentCache()
     ) {
-        self.apiKey = apiKey;
-        self.host = host;
-        self.assignmentLogger = assignmentLogger;
+        self.apiKey = apiKey
+        self.host = host
+        self.assignmentLogger = assignmentLogger
         self.assignmentCache = assignmentCache
-
-        let httpClient = NetworkEppoHttpClient(baseURL: host, apiKey: apiKey, sdkName: sdkName, sdkVersion: sdkVersion);
-        let configurationRequester = ConfigurationRequester(
-            httpClient: httpClient  
-        );
-        self.configurationStore = ConfigurationStore(requester: configurationRequester);
+        
+        let httpClient = NetworkEppoHttpClient(baseURL: host, apiKey: apiKey, sdkName: "sdkName", sdkVersion: "sdkVersion")
+        let configurationRequester = ConfigurationRequester(httpClient: httpClient)
+        self.configurationStore = ConfigurationStore(requester: configurationRequester)
     }
-
-    public func load() async throws {
+    
+    public static func configure(
```

Comment:
we call this "init" in the other SDKs. Is "configure" a swift convention or just trying new lingo?

Reply:
+1 to keeping it consistent across languages

Reply:
We cannot use `init` as it is a reserved word in Swift for the class constructor: https://docs.swift.org/swift-book/documentation/the-swift-programming-language/initialization/

Another note: the `configure` method while it sets the desired user options does NOT load the RAC/UFC data. Our documentation suggestions callers wrap `loadIfDesired` in a `Task` but I will investigate if this is feasible to happen within the SDK.